tfhe/src/core_crypto/experimental/algorithms/lwe_shrinking_keyswitch_key_generation.rs

```diff
@@ -0,0 +1,106 @@
+//! Module containing primitives pertaining to [`LWE shrinking keyswitch keys
+//! generation`](`crate::core_crypto::entities::LweKeyswitchKey#key-switching-key`).
+
+use crate::core_crypto::algorithms::generate_lwe_keyswitch_key;
+use crate::core_crypto::commons::generators::EncryptionRandomGenerator;
+use crate::core_crypto::commons::math::random::{Distribution, Uniform};
+use crate::core_crypto::commons::parameters::*;
+use crate::core_crypto::commons::traits::*;
+use crate::core_crypto::entities::LweSecretKey;
+use crate::core_crypto::experimental::commons::parameters::LweSecretKeySharedCoefCount;
+use crate::core_crypto::experimental::entities::{
+    LweShrinkingKeyswitchKey, LweShrinkingKeyswitchKeyOwned,
+};
+
+pub fn generate_lwe_shrinking_keyswitch_key<
+    Scalar,
+    NoiseDistribution,
+    InputKeyCont,
+    OutputKeyCont,
+    KSKeyCont,
+    Gen,
+>(
+    input_lwe_sk: &LweSecretKey<InputKeyCont>,
+    output_lwe_sk: &LweSecretKey<OutputKeyCont>,
+    lwe_keyswitch_key: &mut LweShrinkingKeyswitchKey<KSKeyCont>,
+    noise_distribution: NoiseDistribution,
+    generator: &mut EncryptionRandomGenerator<Gen>,
+) where
+    Scalar: Encryptable<Uniform, NoiseDistribution>,
+    NoiseDistribution: Distribution,
+    InputKeyCont: Container<Element = Scalar>,
+    OutputKeyCont: Container<Element = Scalar>,
+    KSKeyCont: ContainerMut<Element = Scalar>,
+    Gen: ByteRandomGenerator,
+{
+    assert!(
```

Comment:
assert_eq

Reply:
removed as it's checked by ksk generation